src/circuit/layouter.rs

```diff
@@ -160,7 +238,6 @@ impl<'r, 'a, C: Chip, CS: Assignment<C::Field> + 'a> RegionLayouter<C>
             self.layouter.regions[self.region_index] + offset,
             to,
         )?;
-        self.row_count = cmp::max(self.row_count, offset);
```

Comment:
Note that this PR also fixes an off-by-one error that existed here (which was causing the last row of a region to be overwritten by the next region).